apiserver/facade/interface.go

```diff
@@ -31,6 +31,9 @@ type Facade interface{}
 // Factory is a callback used to create a Facade.
 type Factory func(stdCtx context.Context, facadeCtx Context) (Facade, error)
 
+// ModelFactory is a callback used to create a Facade.
+type ModelFactory func(stdCtx context.Context, facadeCtx ModelContext) (Facade, error)
```

Comment:
These names need work. `ModelFactory` could imply that it's for a particular model rather than providing the ability to work on _other_ models. Same with `ModelContext`.

The comments across the different versions are inadequate for distinguishing them too.

I think the nomenclature should allude to model specific context/factory and multi-model context/factory.